### ISSUE TITLE
Precompressed sitemaps have incorrect HTTP headers

### ISSUE THREAD

**Issue by benschwarz** [mdn/yari#8497]
### Summary

Some of the sitemap files generated for the MDN site use gzip for compression. Unfortunately the headers returned from Cloudfront do not include `Content-Encoding` headers, so clients are not able to know that they must decompress the sitemap.

### URL

https://developer.mozilla.org/sitemaps/en-us/sitemap.xml.gz

### Reproduction steps

1. Run curl -I https://developer.mozilla.org/sitemaps/en-us/sitemap.xml.gz

### Expected behavior

`Content-Encoding` header should be present

### Actual behavior

- No content encoding header is returned
- `Content-Type` reports the resource as `application/xml`

### Device

Desktop

### Browser

Chrome

### Browser version

Stable

### Operating system

Mac OS

### Screenshot

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/924/227758727-26ced6c0-9d0f-4afa-ab18-bda06fb0343e.png">

### Anything else?

I'd be prepared to work on a fix for this, though I wasn't able to locate where to do it 😅

If a contributor can point me in the right direction I'd be happy to submit a PR to correct the behavior! 

Thanks for MDN 🦖

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/orgs/mdn/discussions/categories/platform).

**Comment by caugner** [comment]
Thank you for bringing this to our attention! 👋 

This is actually a limitation/bug of our current deployer implementation, which _could_ use the `content-encoding` returned by `mimetypes.guess_type()`, but just doesn't:

https://github.com/mdn/yari/blob/92e025701df287df3f3553c74b32fd9a23cef2f0/deployer/src/deployer/upload.py#L180-L182

Anyhow, this deployer is being superseded by https://github.com/mdn/yari/pull/8366, and I have added a special handling for `sitemap.xml.gz` files now (65393dbe674b1f22d0640c86ba7408bc990a53d0), as Google Cloud Storage itself only returns `Content-Type` without `Content-Encoding`, see: https://storage.googleapis.com/fiji-mdn-content/sitemaps/en-us/sitemap.xml.gz 🤷

**Comment by benschwarz** [comment]
Thanks for working to resolve this issue @caugner. 

I just re-tested the same URL (using the mechanism described above), and the `Content-Encoding` header is still omitted: 

```
curl -I https://developer.mozilla.org/sitemaps/en-us/sitemap.xml.gz
```

Yields: 

```
HTTP/2 200
content-type: application/xml
content-length: 97161
date: Sat, 08 Apr 2023 00:51:41 GMT
cache-control: max-age=86400, public
last-modified: Fri, 07 Apr 2023 01:05:50 GMT
etag: "48fe6dd34343ee230c90db0e12026334"
server: AmazonS3
x-frame-options: DENY
x-xss-protection: 1; mode=block
x-content-type-options: nosniff
strict-transport-security: max-age=63072000
vary: Accept-Encoding
x-cache: Hit from cloudfront
via: 1.1 0c97e6586361cfaa664cfcbf520d2336.cloudfront.net (CloudFront)
x-amz-cf-pop: MEL52-P1
alt-svc: h3=":443"; ma=86400
x-amz-cf-id: fp3qLr7GuLY4BJ-IHWNpVkplxo-z02RE253EUKEQwZucUh5_UitqkQ==
age: 57
```

Not sure if we're just waiting on the next deploy to go out or now, but thought I'd raise anyway.

**Comment by yin1999** [comment]
Hi @benschwarz

We build content [every midnight](https://github.com/mdn/yari/actions/runs/4642275536/workflow#L20), so there would not have any content changes before `00:00 UTC+0`. You could try this as the new content has been deployed now.

```bash
allo@debian:~$ curl -I https://developer.mozilla.org/sitemaps/en-us/sitemap.xml.gz
HTTP/2 200 
content-type: application/xml
content-length: 95375
date: Sat, 08 Apr 2023 01:32:22 GMT
cache-control: max-age=86400, public
content-encoding: gzip
last-modified: Sat, 08 Apr 2023 00:53:45 GMT
etag: "bcce4806ed215f6a52013e5065c8f379"
server: AmazonS3
x-frame-options: DENY
x-xss-protection: 1; mode=block
x-content-type-options: nosniff
strict-transport-security: max-age=63072000
x-cache: Miss from cloudfront
via: 1.1 f7aba4a0337c5f98c4703e2b10f1940a.cloudfront.net (CloudFront)
x-amz-cf-pop: FRA60-P4
alt-svc: h3=":443"; ma=86400
x-amz-cf-id: 8sL-D1FaMuKl5RzrtIalZ5zydATi1W6OWXIrswV9yTcXddylLuFpxQ==
```

**Comment by benschwarz** [comment]
Thanks for the confirmation @yin1999 🙏 

I can see that the header has updated this last hour. LGTM 👍
